lib/pure/collections/sharedtables.nim

```diff
@@ -60,17 +60,24 @@ template withLock(t, x: untyped) =
 
 template withValue*[A, B](t: var SharedTable[A, B], key: A,
                           value, body: untyped) =
-  ## retrieves the value at `t[key]`.
+  ## Retrieves the value at `t[key]`.
   ## `value` can be modified in the scope of the `withValue` call.
-  ##
-  ## .. code-block:: nim
-  ##
-  ##   sharedTable.withValue(key, value) do:
-  ##     # block is executed only if `key` in `t`
-  ##     # value is threadsafe in block
-  ##     value.name = "username"
-  ##     value.uid = 1000
-  ##
+  runnableExamples:
+    var table: SharedTable[string, string]
+    init(table)
+
```

Comment:
unrelated to this PR but IMO we need a toSharedTable, analog to toTable + friends, eg this would simplify as:
```nim
var table = {"a": "x", "b": "y", "c": "z"}.toSharedTable
```